test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -0,0 +1,360 @@
+        cls.services = testClient.getParsedTestDataConfig()
+
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
+        cls.hypervisor = testClient.getHypervisorInfo()
+        cls.services['mode'] = cls.zone.networktype
+        cls.hostConfig = cls.config.__dict__["zones"][0].__dict__["pods"][0].__dict__["clusters"][0].__dict__["hosts"][
+            0].__dict__
+
+        # Set Zones and disk offerings
+        cls.services["small"]["zoneid"] = cls.zone.id
+
+        cls.services["iso1"]["zoneid"] = cls.zone.id
+
+        cls.services["virtual_machine"]["zoneid"] = cls.zone.id
+
+        # Create an account, network, and IP addresses
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            domainid=cls.domain.id
+        )
+        cls.service_offering = ServiceOffering.create(
+            cls.apiclient,
+            cls.services["service_offerings"]["small"]
+        )
+
+        cls.cleanup = [
+            cls.account,
+            cls.service_offering
+        ]
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.apiclient = super(TestAddConfigtoDeployVM, cls).getClsTestClient().getApiClient()
+            # Clean up, terminate the created templates
+            cleanup_resources(cls.apiclient, cls.cleanup)
+
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.hypervisor = self.testClient.getHypervisorInfo()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+        """
+        Set EnableAdditionalData to true
+        """
+        updateConfigurationCmd = updateConfiguration.updateConfigurationCmd()
+        updateConfigurationCmd.name = "enable.additional.vm.configuration"
+        updateConfigurationCmd.value = "true"
+        updateConfigurationCmd.scopename = "account"
+        updateConfigurationResponse = self.apiclient.updateConfiguration(updateConfigurationCmd)
+        self.debug("updated the parameter %s with value %s" % (
+        updateConfigurationResponse.name, updateConfigurationResponse.value))
+
+    def tearDown(self):
+        pass
+
+    # Compare XML Element objects
+    def elements_equal(self, e1, e2):
+        if e1.tag != e2.tag:
+            return False
+        if e1.attrib != e2.attrib:
+            return False
+        if len(e1) != len(e2):
+            return False
+        return all(self.elements_equal(c1, c2) for c1, c2 in zip(e1, e2))
+
+    def destroy_vm(self, id):
+        cmd = destroyVirtualMachine.destroyVirtualMachineCmd()
+        cmd.expunge = True
+        cmd.id = id
+        return self.apiclient.destroyVirtualMachine(cmd)
+
+    def deploy_vm(self, hypervisor, extra_config=None):
```

Comment:
VMs are left over after the execution, we'll need to append the newly created VM to cls.cleanup